axelor-supplychain/src/main/resources/views/Menu.xml

```diff
@@ -277,5 +277,19 @@
     <view type="grid" name="stock-move-invoice-blocking-reason-grid"/>
     <view type="form" name="stock-move-invoice-blocking-reason-form"/>
   </action-view>
+
+  <menuitem name="sc-root-sale-order-lines-to-invoice"
+    title="Mass sale order line invoicing" parent="sc-root-sale" order="601"
+    action="sc.root.sale.order.lines.to.invoice"/>
+
+  <action-view name="sc.root.sale.order.lines.to.invoice"
+    title="Mass sale order line invoicing" model="com.axelor.apps.sale.db.SaleOrderLine">
+    <view type="grid" name="sale-order-line-to-invoice-grid"/>
+    <view type="form" name="sale-order-line-form"/>
+    <domain>
+      (self.invoiced = false OR self.amountInvoiced &lt; self.qty * self.price) AND
```

Comment:
"invoiced" is used only for special process (it will probably be removed in current or next version), and directly checking the amountInvoiced does not manage all cases, so please instead use invoicingState
```suggestion
      (self.invoicingState = 1) AND
```